django/thunderstore/frontend/api/experimental/views/frontpage.py

```diff
@@ -0,0 +1,104 @@
+from typing import Optional
+
+from django.db.models import Sum
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from thunderstore.community.models.community import Community
+from thunderstore.frontend.api.experimental.serializers.views import (
+    FrontPageContentSerializer,
+)
+from thunderstore.repository.models import Package, PackageVersion
+
+
+class FrontPageApiView(APIView):
+    """
+    Return information required to render the site's front page.
+    """
+
+    serializer_class = FrontPageContentSerializer
+    permission_classes = []
+
+    @swagger_auto_schema(
+        responses={200: FrontPageContentSerializer()},
+        operation_id="experimental.frontend.frontpage",
+    )
+    def get(self, request, *args, **kwargs):
+        community_objs = Community.objects.listed().prefetch_related(
+            "package_listings__package__versions",
+            "sites",
+        )
+        communities = [
+            {
+                "downloadCount": get_community_downloads(c),
+                "imageSrc": get_site_image(c),
+                "modCount": get_community_mods(c),
+                "name": c.name,
+            }
+            for c in community_objs
+        ]
+        communities.sort(key=lambda c: c["modCount"], reverse=True)
```

Comment:
If you annotate the package count, you can use a queryset `.order_by("-package_count")` here as well.